Add state object detail to numeric state

mekaneck opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
The numeric state trigger populates an object called `state` but this was not in the documentation previously. This change makes the availability of that object explicit and also provides a link to the relevant state object documentation.


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->
None

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

mekaneck committed:
Add state object detail to numeric state

The numeric state trigger populates an object called `state` but this was not in the documentation previously. This change makes it explicit and also provides a link to the relevant state object documentation.

Didgeridrew commented on source/_docs/automation/trigger.markdown:
```diff
@@ -257,7 +257,7 @@ automation:
 {% endraw %}
 
-More dynamic and complex calculations can be done with `value_template`.
+More dynamic and complex calculations can be done with `value_template`. When this option is specified an object named `state` will be available which is populated with the state object of the entity specified by `entity_id`. See the [state objects documentation](/docs/configuration/state_object/) for details on how to access the contents of this object. 
```
To mirror documentation about similar variables you might want to go with:

```
More dynamic and complex calculations can be done with `value_template`. The variable 'state' is the [state object](https://www.home-assistant.io/docs/configuration/state_object) of the entity specified by `entity_id`.
```

It would also be helpful to add or modify one of the example configurations to include the use of `state.state`.

mekaneck replied:
Updated with the more concise explanation, and also added and example with `state.state`

mekaneck committed:
Update trigger.markdown

User more concise explanation of the state object in the numeric trigger, and added an example using `state.state`

mekaneck committed:
Update trigger.markdown

Added explanation to examples that use `state`

Troon reviewed (approved):
Looks good to me.

c0ffeeca7 reviewed (approved):
Thank you, @mekaneck :+1:
